pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryConfig.java

```diff
@@ -22,20 +22,43 @@
  * Configuration for setting up query runtime.
  */
 public class QueryConfig {
-  public static final long DEFAULT_TIMEOUT_NANO = 10_000_000_000L;
-
+  /**
+   * Configuration for mailbox data block size
+   */
   public static final String KEY_OF_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES = "pinot.query.runner.max.msg.size.bytes";
   public static final int DEFAULT_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES = 16 * 1024 * 1024;
 
+  public static final String KEY_OF_MAILBOX_TIMEOUT_MS = "pinot.query.runner.mailbox.timeout.ms";
+  public static final long DEFAULT_MAILBOX_TIMEOUT_MS = 10_000L;
```

Comment:
Might have already been discussed before, but do we need a separate timeout for mailbox?

Is there a scenario where we'd want one timeout for mailbox and another for the query?

Reply:
yes but since we dont have that configuration hook up I will defer that to the next PR (see description above already mentioned this)

Reply:
bare in mind that even this only allows configuration using the KEY during config load time, there's still no way to support this on a per-query basis, I added per-query, at-runtime as "maybe" in the TODO list